homeassistant/components/weather/__init__.py

```diff
@@ -381,6 +387,68 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
                     cls.__name__,
                     report_issue,
                 )
+        if any(
+            method in cls.__dict__ for method in ("_attr_forecast", "forecast")
+        ) and not any(
+            method in cls.__dict__
+            for method in (
+                "async_forecast_daily",
+                "async_forecast_hourly",
+                "async_forecast_twice_daily",
+            )
+        ):
+            cls.__weather_legacy_forecast = True
+
+    @callback
+    def add_to_platform_start(
+        self,
+        hass: HomeAssistant,
+        platform: EntityPlatform,
+        parallel_updates: asyncio.Semaphore | None,
+    ) -> None:
+        """Start adding an entity to a platform."""
+        super().add_to_platform_start(hass, platform, parallel_updates)
+        _reported_forecast = False
+        if self.__weather_legacy_forecast and not _reported_forecast:
+            module = inspect.getmodule(self)
+            if module and module.__file__ and "custom_components" in module.__file__:
+                report_issue = "report it to the custom integration author."
+            else:
+                report_issue = (
+                    "create a bug report at "
+                    "https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue"
+                )
```

Comment:
There are no core integrations left to upgrade, we can simplify the logic here to return early without logging or creating an issue if the integration is not a custom integration.